File: problem206.php
<?php
	$min = 101010103;
	$max = 138902663;

	$i = $min;
	
	function check ($num) {
		$square = bcmul($num, $num);
		
		if($square[0] == 1 && $square[2] == 2 && $square[4] == 3 && $square[6] == 4 && $square[8] == 5 &&
			$square[10] == 6 && $square [12] == 7 && $square[14] == 8 && $square[16] == 9) {
					return 1;
			} else {
				return 0;	
			}
	}
	
	while($i <= $max) {
		if(check($i)) break;
		if($i%10 == 3){
			$i = $i + 4;
		} else {
			$i = $i + 6;
		}
	}

	$i = bcmul("10", $i);
	
	$myFile = "problem206.txt";
	$fh = fopen($myFile, 'w') or die("can't open file");
	fwrite($fh, $i);
	fclose($fh);
?>

File: problem206.c
#include <stdio.h>

long long power (long long base, int exp)
{
	long long result = 1;

	while (exp > 0) {
		result = result * base;
		exp--;
	}

	return result;
}

/* Returns the PLACE-th digit of NUM.
 */
 int getDigit(long long num, int place) {
	num = num%power(10, place);
	num = num - (num%power(10,place-1));
	num = num/power(10,place-1);

	return num;
}

int check (long long num) {
	long long square = num * num;

	if(square%10 == 9 && getDigit(square, 3) == 8 && getDigit(square, 5) == 7 &&
		getDigit(square, 7) == 6 && getDigit(square, 9) == 5 && getDigit(square, 11) == 4 &&
		getDigit(square, 13) == 3 && getDigit(square, 15) == 2 && getDigit(square, 17) == 1) {
			return 1;
		} else {
			return 0;
		}
}

int main ( ) {
	long long min = 101010103;
	long long max = 138902663;

	while(min <= max) {
		if(check(min)) break;
		if(min%10 == 3) {
			min = min + 4;
		} else {
			min = min + 6;
		}
	}

	min = min*10;

	printf("%lld", min);

	return 0;

}
